src/hooks/const.ts

```diff
@@ -0,0 +1,10 @@
+import { useMemo } from "react";
+
+/**
+ * This function is used to create a constant value that will not be reinitialized on re-renders.
+ * @param input
+ */
+export default function useConst<T>(input: T) {
```

Comment:
Interesting...

Reply:
This is used in useSearchParamsState, as it has a initialState props which should be a const config object, but developer would likely using the hook like useSearchParamsState({test: 1}) which generate a new object every time and lead to state change.